zerver/tests/test_outgoing_webhook_system.py

```diff
@@ -225,6 +226,34 @@ def test_request_exception(self) -> None:
         assert bot_user.bot_owner is not None
         self.assertEqual(bot_owner_notification.recipient_id, bot_user.bot_owner.recipient_id)
 
+    def test_jsonable_exception(self) -> None:
+        bot_user = self.example_user("outgoing_webhook_bot")
+        mock_event = self.mock_event(bot_user)
+        service_handler = GenericOutgoingWebhookService("token", bot_user, "service")
+
+        # Widget content is being decoded twice, once in process_success_response along with response
+        # and again in check_messages leading to always cause an error
```

Comment:
Re-reading this, I'm having a hard time following this message.  What is `check_messages`?  I don't see a function named that.

Can you help me understand better what you're trying to say here?

Reply:
We can prevent the `JsonableError` from being raised if we encode (`json.dumps`) `widget_content` in [`process_success_response`](https://github.com/zulip/zulip/blob/c693ae8982c12962e2fb510a18a947f16b9af640/zerver/lib/outgoing_webhook.py#L270), as this error is being raised because `widget_content` is being decoded twice, once in [`process_success_response`](https://github.com/zulip/zulip/blob/c693ae8982c12962e2fb510a18a947f16b9af640/zerver/lib/outgoing_webhook.py#L274) and once more in [`check_message`](https://github.com/zulip/zulip/blob/c693ae8982c12962e2fb510a18a947f16b9af640/zerver/lib/actions.py#L2448).